src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -512,81 +518,81 @@ public void testSamReaderFromSeekableStream() throws IOException {
         Assert.assertEquals(countRecords(reader), 10);
     }
 
-
     @Test
     public void testSamReaderFromURL() throws IOException {
         final String samFilePath = new File(TEST_DATA_DIR, "unsorted.sam").getAbsolutePath();
         final URL samURL = new URL("file://" + samFilePath);
         final SamReaderFactory factory = SamReaderFactory.makeDefault()
                 .validationStringency(ValidationStringency.SILENT);
-        final SamReader reader = factory.open(SamInputResource.of(samURL));
-        Assert.assertEquals(countRecords(reader), 10);
+        try (final SamReader reader = factory.open(SamInputResource.of(samURL))) {
+            Assert.assertEquals(countRecords(reader), 10);
+        }
     }
 
-    @Test(expectedExceptions=SAMFormatException.class)
+    @Test(expectedExceptions = SAMFormatException.class)
     public void testSamReaderFromMalformedSeekableStream() throws IOException {
         // use a bogus (.bai file) to force SamReaderFactory to fall through to the
         // fallback code that assumes a SAM File when it can't determine the
         // format of the input, to ensure that it results in a SAMFormatException
         final File samFile = new File(TEST_DATA_DIR, "cram_with_bai_index.cram.bai");
         final SamReaderFactory factory = SamReaderFactory.makeDefault()
                 .validationStringency(ValidationStringency.SILENT);
-        final SamReader reader = factory.open(
-                SamInputResource.of(new SeekableFileStream(samFile)));
-        countRecords(reader);
+        try (final SamReader reader = factory.open(
+                SamInputResource.of(new SeekableFileStream(samFile)))) {
+            countRecords(reader);
+        }
     }
 
-    @Test(singleThreaded = true, groups="unix")
-    public void testWriteAndReadFromPipe() throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    @DataProvider()
```

Comment:
I think turning this into a data provider is _less_ clear that just having `for x in Arrays.asList(true, false)` in the test.